spring-integration-jmx/src/main/java/org/springframework/integration/monitor/IntegrationMBeanExporter.java

```diff
@@ -770,6 +784,10 @@ private void registerChannels() {
 				enabled = smartMatch(this.enabledStatsPatterns, name);
 				if (enabled != null) {
 					monitor.enableStats(enabled);
+					metrics.setFullStatsEnabled(enabled);
```

Comment:
RIght, but we'd have to do a null check again later; so it's a choice of a second null check or an unnecessary setter call; I chose the latter (cleaner).